packages/entities/entities-plugins/docs/plugin-select.md

```diff
@@ -0,0 +1,190 @@
+
+  - `getCustomEditRoute`:
+    - type: `(plugin: string) => RouteLocationRaw`
+    - required: `false`
+    - default: `undefined`
+    - A function that returns the route for editing a custom plugin.
+
+  - `workspace`:
+    - type: `string`
+    - required: `true`
+    - default: `undefined`
+    - *Specific to Kong Manager*. Name of the current workspace.
+
+  - `controlPlaneId`:
+    - type: `string`
+    - required: `true`
+    - default: `undefined`
+    - *Specific to Konnect*. Name of the current control plane.
+
+  - `entityId`:
+    - type: `string`
+    - required: `false`
+    - default: `null`
+    - Current entity id if the PluginSelect is being launched from the plugins tab on a consumer, consumer group, gateway service, or route detail page.
+
+  - `entityType`:
+    - type: `'consumers' | 'consumer_groups' | 'services' | 'routes'`
+    - required: `false`
+    - default: `null`
+    - Current entity type if the PluginSelect is being launched from the plugins tab on a consumer, consumer group, gateway service, or route detail page.
+
+The base konnect or kongManger config.
+
+#### `canCreateCustom`
+
+- type: `Function as PropType<() => boolean | Promise<boolean>>`
+- required: `false`
+- default: `async () => true`
+
+A synchronous or asynchronous function, that returns a boolean, that evaluates if the user can create a new custom plugin.
+
+#### `canDeleteCustom`
+
+- type: `Function as PropType<(row: object) => boolean | Promise<boolean>>`
+- required: `false`
+- default: `async () => true`
+
+A synchronous or asynchronous function, that returns a boolean, that evaluates if the user can delete a given custom plugin.
+
+#### `canEditCustom`
+
+- type: `Function as PropType<(row: object) => boolean | Promise<boolean>>`
+- required: `false`
+- default: `async () => true`
+
+A synchronous or asynchronous function, that returns a boolean, that evaluates if the user can edit a given custom plugin.
+
+#### `noRouteChange`
+
+- type: `boolean`
+- required: `false`
+- default: `false`
+
+If true, let consuming component handle event when clicking on a plugin. Used in conjunction with `@plugin-clicked` event.
+
+#### `onlyAvailablePlugins`
+
+- type: `boolean`
+- required: `false`
+- default: `false`
+
+Checks the kong config plugins.available_on_server and if true, then it will not show plugins from PluginMeta that are outside of the available_on_server array.
+
+#### `ignoredPlugins`
+
+- type: `string[]`
+- required: `false`
+- default: '[]'
+
+An array of the plugin names. These are plugins that should not be displayed.
+
+#### `disabledPlugins`
+
+- type: `DisabledPlugin`
+- required: `false`
+- default: `{}`
+
+Plugins that should be disabled and their disabled messages.
+Example:
+
+```json
+{
+  'acl': 'ACL is not supported for this entity type',
+}
+```
+
+#### `pluginsPerRow`
+
+- type: `number`
+- required: `false`
+- default: `4`
+
+Number of plugins to always have visible (never will be collapsed).
```

Comment:
I think we can't do that. We can't use standard wrapping for this because what this actually is, is the number of plugins in the "always visible" section of the KCollapse and I don't think the logic to determine that on the fly would be worth it, it isn't the same thing as the number of plugins to display in a row before wrapping. We would end up needing a staging area like we have for KMultiselect to determine how many plugins we can fit in a single row and then have to do this for each plugin section.